frontend/src/scenes/heatmaps/HeatmapsBrowser.tsx

```diff
@@ -76,82 +84,144 @@ function ExportButton({
 }
 
 function UrlSearchHeader({ iframeRef }: { iframeRef?: React.MutableRefObject<HTMLIFrameElement | null> }): JSX.Element {
-    const logic = heatmapsBrowserLogic()
-
-    const { browserUrlSearchOptions, browserUrl, isBrowserUrlValid, replayIframeData, hasValidReplayIframeData } =
-        useValues(logic)
-    const { setBrowserSearch, setBrowserUrl, setReplayIframeData, setReplayIframeDataURL } = useActions(logic)
+    const {
+        browserUrlSearchOptions,
+        browserUrl,
+        isBrowserUrlValid,
+        replayIframeData,
+        hasValidReplayIframeData,
+        browserSearchTerm,
+        displayUrl,
+    } = useValues(heatmapsBrowserLogic)
+    const { setBrowserSearch, setBrowserUrl, setReplayIframeData, setReplayIframeDataURL, setDisplayUrl } =
+        useActions(heatmapsBrowserLogic)
 
     const placeholderUrl = browserUrlSearchOptions?.[0] ?? 'https://your-website.com/pricing'
 
     return (
-        <div className="flex-none md:flex justify-between items-center gap-2 w-full">
-            <div className="flex-1 min-w-0">
+        <>
+            <div className="flex-none md:flex justify-between items-end gap-2 w-full">
+                <div className="flex gap-2 flex-1 min-w-0">
+                    <div className="flex-1">
+                        {hasValidReplayIframeData ? (
+                            <LemonInput
+                                value={replayIframeData?.url}
+                                onChange={(s) => setReplayIframeDataURL(s)}
+                                className="truncate"
+                                size="small"
+                            />
+                        ) : (
+                            <>
+                                <div className="relative">
+                                    <LemonLabel>Display URL</LemonLabel>
+                                    <div className="text-xs text-muted mb-1">
+                                        Enter a working URL from your site for iframe preview
+                                    </div>
+                                    <div className="flex gap-2">
+                                        <LemonInput
+                                            size="small"
+                                            placeholder={`e.g. ${placeholderUrl}`}
+                                            value={displayUrl || browserSearchTerm || ''}
+                                            onChange={(value) => {
+                                                setBrowserSearch(value)
+                                                setDisplayUrl(value || null)
+
+                                                // Copy the same URL to heatmap data URL
+                                                setBrowserUrl(value || null)
```

Comment:
i wonder if setDisplay and setBrowser are the right names here

setDisplay and setData?